meson.build

```diff
@@ -0,0 +1,470 @@
+src_xdp_utils = files('src/xdp-utils.c', 'src/xdp-utils.h')
+subdir('document-portal')
+
+#built_src_permission_store = gnome.gdbus_codegen('permission-store-dbus',
+#    sources: 'data/org.freedesktop.impl.portal.PermissionStore.xml',
+#    interface_prefix: 'org.freedesktop.impl.portal',
+#    namespace: 'Xdg',
+#    autocleanup: 'none',
+#    )
+#src_db = files(
+#    'document-portal/permission-db.c',
+#    'document-portal/permission-db.h',
+#    'document-portal/gvdb/gvdb-reader.h',
+#    'document-portal/gvdb/gvdb-format.h',
+#    'document-portal/gvdb/gvdb-reader.c',
+#    'document-portal/gvdb/gvdb-builder.h',
+#    'document-portal/gvdb/gvdb-builder.c',
+#)
+#src_xdg_permission_store = files(
+#    'src/xdp-utils.c',
+#    'src/xdp-utils.h',
+#    'document-portal/permission-store.c',
+#    'document-portal/xdg-permission-store.c',
+#    'document-portal/xdg-permission-store.h',
+#    ) + src_db + built_src_permission_store
+#
+#xdg_permission_store = executable('xdg-permission-store',
+#    src_xdg_permission_store,
+#    dependencies: deps_xdg_desktop_portal,
+#    include_directories: incs_xdg_destop_portal,
+#    install: true,
+#    install_dir: dir_libexecdir
+#    )
+#
+#built_src_document_portal = gnome.gdbus_codegen('document-portal-dbus',
+#    sources: [
+#        'data/org.freedesktop.portal.Documents.xml',
+#        'data/org.freedesktop.portal.FileTransfer.xml',
+#    ],
+#    interface_prefix: 'org.freedesktop.portal',
+#    namespace: 'XdpDbus',
+#    autocleanup: 'none',
+#    )
+#src_xdg_document_portal = files(
+#    'src/xdp-utils.c',
+#    'src/xdp-utils.h',
+#    'document-portal/document-portal.h',
+#    'document-portal/document-portal.c',
+#    'document-portal/file-transfer.h',
+#    'document-portal/file-transfer.c',
+#    'document-portal/document-enums.h',
+#    'document-portal/document-store.h',
+#    'document-portal/document-store.c',
+#    'document-portal/document-portal-fuse.h',
+#    'document-portal/document-portal-fuse.c',
+#    ) + src_db + built_src_document_portal + built_src_permission_store
+#
+#xdg_document_portal = executable('xdg-document-portal',
+#    src_xdg_document_portal,
+#    dependencies: [deps_xdg_desktop_portal, dep_fuse3],
+#    include_directories: incs_xdg_destop_portal,
+#    install: true,
+#    install_dir: dir_libexecdir
+#    )
+
+###### documentation
+
+if build_docbook
+    xslfile = files('doc/xmlto-config.xsl')
+
+    portal_docs_xml = configure_file(
+        input: 'doc/portal-docs.xml.in',
+        output: '@BASENAME@',
+        configuration: {'VERSION': meson.project_version()},
+    )
+
+    custom_target('html',
+            build_by_default: true,
+            depends: [built_src_portal, built_src_impl],
+            input: portal_docs_xml,
+            output: 'portal-docs.html',
+            command: [prog_xmlto, get_option('xmlto-flags'), 'xhtml-nochunks', '-m', xslfile, '@INPUT@'])
+endif
+
+# FIXME: coverage testing  is missing
```

Comment:
Meson automatically provides the `b_coverage` option, so we don't need to write any code to make it possible to track test coverage.